sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystQl.scala

```diff
@@ -667,6 +667,8 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
           UnresolvedAttribute(nameParts :+ cleanIdentifier(attr))
         case other => UnresolvedExtractValue(other, Literal(cleanIdentifier(attr)))
       }
+    case Token("TOK_SUBQUERY_EXPR", Token("TOK_SUBQUERY_OP", Nil) :: subquery :: Nil) =>
+      ScalarSubquery(nodeToPlan(subquery))
```

Comment:
This might sound excedingly dumb but I cannot find `ScalarSubquery' or `SubqueryExpression`. Are they already in the code base? Or did you create branch on top of another branch?